feat: add capabilities to app.getLoginItemSettings() and app.setLoginItemSettings() (#24494)
electron/electron: docs/api/app.md

  should restore the state from the previous session. This indicates that the
  app should restore the windows that were open the last time the app was
  closed. This setting is not available on [MAS builds][mas-builds].
* `executableWillLaunchAtLogin` Boolean _Windows_ - `true` if app is set to open at login and its run key is not deactivated. This differs from `openAtLogin` as it ignores the `args` option, this property will be true if the given executable would be launched at login with **any** arguments.
* `launchItems` Object[] _Windows_
  * `name` String _Windows_ - name value of a registry entry.
  * `path` String _Windows_ - The executable to an app that corresponds to a registry entry.
  * `args` String[] _Windows_ - the command-line arguments to pass to the executable.

>   * `scope` String _Windows_ - one of `user` or `machine`. Indicates whether the registry entry is under `HARD_KEY_CURRENT USER` or `HARD_KEY_LOCAL_MACHINE`.

Reviewer: ```suggestion
  * `scope` String _Windows_ - one of `user` or `machine`. Indicates whether the registry entry is under `HKEY_CURRENT USER` or `HKEY_LOCAL_MACHINE`.
```

  should restore the state from the previous session. This indicates that the
  app should restore the windows that were open the last time the app was
  closed. This setting is not available on [MAS builds][mas-builds].

> * `executableLaunchAtLogin` Boolean _Windows_ - `true` if app is set to open at login and its run key is not deactivated. This differs from `openAtLogin` as it ignores the `args` option, this property will be true if the given executable would be launched at login with **any** arguments.

Reviewer 2: `executableWillLaunchAtLogin`

Reviewer 2: + change all other occurrences please

> * `launchItems` Object[] _Windows_ - an array of launch items populated from the user's key registry

Reviewer: " .. items populated from the user and machine registry hive"

> * `launchItems` Object[] _Windows_ - an array of launch items populated from the user and machine registry hive

Reviewer 2: Please document the properties on this `Object`

  * `args` String[] (optional) _Windows_ - The command-line arguments to pass to
    the executable. Defaults to an empty array. Take care to wrap paths in
    quotes.

>   * `enabled` Boolean (optional) _Windows_ - `true` if the app registry key is startup approved.

Reviewer: "if the app registry key is startup approved and therefore shows as enabled in Task Manager and Windows settings

    Defaults to `true`.

>   * `name` String (optional) _Windows_ - value name to write into registry. Defaults to the app's process name.

Reviewer: "Defaults to the apps AppUserModelId"

  * `args` String[] (optional) _Windows_ - The command-line arguments to pass to
    the executable. Defaults to an empty array. Take care to wrap paths in
    quotes.

>   * `enabled` Boolean (optional) _Windows_ - `true` if the app registry key is startup approved and therefore shows as `enabled` in Task Manager and Windows settings.

Reviewer: ```suggestion
  * `enabled` Boolean (optional) _Windows_ - `true` will change the startup approved registry key and  `enabled`/`disable` the App in Task Manager and Windows settings.
```
